var/spack/repos/builtin/packages/py-pytest/package.py

```diff
@@ -53,6 +54,7 @@ class PyPytest(PythonPackage):
         depends_on("py-setuptools-scm", when="@3.1:")
 
     with default_args(type=("build", "run")):
+        depends_on("python@3.8:3.12", when="@8.3.4")
```

Comment:
Where do you see this upper bound?

Reply:
Ok thank you, that's good to know. I was going off the list of classifiers in the `pyproject.toml` file. I wasn't sure the best practice here, w.r.t. matching the items in that list. If it's preferable to just leave it open ended I can remove this line.